apollo-router/src/lib.rs

```diff
@@ -225,7 +225,7 @@ impl ConfigurationKind {
             UpdateConfiguration(mut config) => {
                 match trace::try_initialize_subscriber(&config) {
                     Ok(subscriber) => {
-                        config.subscriber = subscriber;
+                        config.subscriber = Some(subscriber);
                     }
                     Err(err) => {
                         tracing::error!("Could not initialize tracing subscriber: {}", err,)
```

Comment:
Suggestion: Perhaps at this point we should set the default subscriber (that for None)?
It seems strange that regardless of configuration the default subscriber is configured, but not in the case that the config has an error.

Reply:
I was wondering about that too. A default subscriber is created to fill the field of the configuration, but if we register it earlier, we do not really need to store it

Reply:
ok so after exploring things a bit more, I am not sure this is something I will be able to solve easily (at least not in the context of this PR).

We set the default subscriber here:

https://github.com/apollographql/router/blob/a296efbfde20c92ca24785b32c3b31b0636306fa/apollo-router/src/main.rs#L89-L97

With the choice for JSON or not JSON logging. A detail here is that `init()` calls [set_global_default](https://docs.rs/tracing/latest/tracing/subscriber/fn.set_global_default.html) which can only be called once, any other call will fail. That subscriber is necessary because we want the ability to log things before the configuration file is loaded.

But with configuration reloads, we cannot call `set_global_default` again, so we create a subscriber from the configuration and pass it explicitely to any new request:

https://github.com/apollographql/router/blob/a296efbfde20c92ca24785b32c3b31b0636306fa/apollo-router/src/warp_http_server_factory.rs#L280

(btw I think that new subscriber does nnot support the json logging, I should check for that)

So I think the subscriber in configuration is still needed for now